libs/cardano-ledger-binary/src/Cardano/Ledger/Binary/Decoding/Decoder.hs

```diff
@@ -833,22 +836,33 @@ decodeSetEnforceNoDuplicates ::
   Ord a =>
   Decoder s a ->
   Decoder s (Set.Set a)
-decodeSetEnforceNoDuplicates decodeElement = do
+decodeSetEnforceNoDuplicates decodeElement =
+  decodeDeduplicate decodeElement Set.member Set.insert -- do
+{-# INLINE decodeSetEnforceNoDuplicates #-}
+
+decodeDeduplicate ::
```

Comment:
This is actually a bad name for this decoder, since deduplication means removal of duplicates, where this decoder will fail on duplicates (that is why it was called enforceNoDuplicates)

I've renamed it in my PR #3801 instead of blocking this PR from being merged.

Reply:
Thanks for fixing it! :+1: :heart: